### PR TITLE
Migrate `ListTile` TextTheme TextStyle references to Material 3

### LINKED ISSUE #89853: [Material 3] Introduce updated text theme
## Proposal

Material 3 comes with an updated/expanded typogrpahy scale to allow for more flexibility.

We will be updating the text theme to support this. Details to come soon

### LINKED ISSUE #102075: Migrate `ListTile` TextTheme TextStyle references to Material 3
### Description

`ListTile` material migration for TextTheme TextStyle

Set `useMaterial3` to true in the ThemeData to use new Material 3 TextTheme TextStyle

### PR DESCRIPTION
**Pull request by TahaTesser**
Updates ListTile [TextTheme] text style references to the Material 3 style names.

Fixes https://github.com/flutter/flutter/issues/102075

Part of: https://github.com/flutter/flutter/issues/91605 (fixes #89853)

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Review by HansMuller — COMMENTED** [review]
I've changed the title of this PR to narrow its scope. In a separate PR we could change the defaults for the ListTile.iconColor - when ThemeData.useMaterial3 = true - to match the Material 3 spec. That means that the default would be based on the theme's color scheme, not IconTheme. We're not trying to simplify making all of the icons in an app have the same color, but we don't want the defaults to be correct for each component type, per the spec.

File: packages/flutter/test/material/list_tile_test.dart [hunk 1]
```diff
@@ -2300,6 +2300,131 @@ void main() {
     expect(description[22], 'minVerticalPadding: 2.0');
     expect(description[23], 'minLeadingWidth: 6.0');
   });
+
+  testWidgets('ListTile inherits IconTheme color', (WidgetTester tester) async {
```
Comment (HansMuller) [inline packages/flutter/test/material/list_tile_test.dart:2304]: This test doesn't belong in this PR

**Review by HansMuller — APPROVED** [review]
LGTM

**Comment by TahaTesser** [comment]
@HansMuller @QuncCccccc 

I think this is caused by this line:

https://github.com/flutter/flutter/pull/101900/files#diff-c4ef7c798c820ecda6cab38be1c2d619d5349dc478889f3a54ebbe07f60c11f2L657

Earlier today I was looking into this new issue https://github.com/flutter/flutter/issues/102121 and I realized `bodyText1` -> `bodyLarge` not `bodyMedium`. (which is in this PR)


https://github.com/flutter/flutter/blob/078deb9230bec918dd5d815092a6c54966ffde6b/packages/flutter/lib/src/material/text_theme.dart#L262-L266

**Comment by TahaTesser** [comment]
After revert is completed, I can file reland PR to match the proper font, which is `bodyLarge` but I am concerned about https://github.com/flutter/flutter/issues/102121, since it looks like spacing is incorrect in this newly added M3 font, 

https://github.com/flutter/flutter/blob/master/packages/flutter/lib/src/material/typography.dart#L613-L614

**Comment by TahaTesser** [comment]
> (maybe a conditional useMaterial3 is needed somewhere).

This sounds great until we're sure M3 fonts are working as expected.

**Comment by TahaTesser** [comment]
While relanding, I can see `bodyLarge` size is bigger than `bodyText1` so reland is still using `bodyMedium`.

**Comment by Gabriel2048** [comment]
Was there a way to disable the default InkWell/ Splash behavior considered?
